### DISCUSSION TITLE
[RFC] Extend Toolkit to scan more directories for queries/mutations, outside of blitz in a monorepo setting

### DISCUSSION THREAD

**Discussion by paulm17** [blitz-js/blitz#3527] [Ideas]
Following on from: https://github.com/blitz-js/blitz/issues/3522 (now closed)

This discussion is for those who use Monorepos such as TurboRepo and have components that exist outside of the blitz install directory.

What's happening now, is that a blitz installation does scan a specific directory but it's within the install directory.  What I would like is to open up a setting, so that an array of directories (submitted by the dev) may be included in this scan.

**Comment by Zeko369** [comment]
Shouldn't be hard to add to Webpack to scan other directories. We could pass the config through `withBlitz` in `next.config.js`

https://github.com/blitz-js/blitz/blob/cacb65d633b1765c1789b123326db02db448ad51/packages/blitz-rpc/src/index-server.ts#L90-L98

**Reply by paulm17** [reply]
Wouldn't this make it a nextjs/webpack setting?  What about when other frameworks are opened up and someone using say remite or vite has this same issue?  🤔

**Reply by paulm17** [reply]
Honestly, since it's a pain not to have this.  Maybe you can just do needs to be done.  Then once [the team] start looking at other frameworks in about a years time.  Someone will revisit this?  😄

**Reply by beerose** [reply]
Yes, I think that we can cover the Next.js use case for now and revisit it later with other frameworks.

**Reply by paulm17** [reply]
@beerose Thank you so much!  This helps me out immensely!  🚀

**Comment by LinnJS** [comment]
This looks sweet. I have had a lot of issues using Blitz with turbo repo specifically when using with Storybook when I have to use `next-transpile-modules`. Overall would love more support for monorepos specifically turborepo.
